Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/debug/FileLoggerTree.kt

```diff
@@ -0,0 +1,86 @@
+package de.rki.coronawarnapp.util.debug
+
+import android.annotation.SuppressLint
+import android.util.Log
+import timber.log.Timber
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.OutputStreamWriter
+
+@SuppressLint("LogNotTimber")
+class FileLoggerTree(private val logFile: File) : Timber.DebugTree() {
+    private var logWriter: OutputStreamWriter? = null
+
+    @SuppressLint("SetWorldReadable")
+    @Synchronized
+    fun start() {
+        if (logWriter != null) return
+
+        logFile.parentFile.mkdirs()
+        if (logFile.createNewFile()) {
+            Log.i(TAG, "File logger writing to " + logFile.path)
+        }
+        if (logFile.setReadable(true, false)) {
+            Log.i(TAG, "Debug run log read permission set")
+        }
+
+        try {
+            logWriter = OutputStreamWriter(FileOutputStream(logFile, true))
+            logWriter!!.write("=== BEGIN ===\n")
+            logWriter!!.write("Logfile: $logFile\n")
+            logWriter!!.flush()
+            Log.i(TAG, "File logger started.")
+        } catch (e: IOException) {
+            e.printStackTrace()
+
+            logFile.delete()
+            if (logWriter != null) logWriter!!.close()
+        }
+    }
+
+    @Synchronized
+    fun stop() {
+        logWriter?.let {
+            logWriter = null
+            try {
+                it.write("=== END ===\n")
+                it.close()
+            } catch (ignore: IOException) {
+            }
+            Log.i(TAG, "File logger stopped.")
+        }
+    }
+
+    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
+        logWriter?.let {
+            try {
+                it.write("${System.currentTimeMillis()}  ${priorityToString(priority)}/$tag: $message\n")
+                it.flush()
```

Comment:
flushing is done after every logline. This happens usually in milliseconds and can penetrate storage memory. I would recommend to buffer in general but I think its ok for a temp solution.